opengever/api/tests/test_protect_dossier.py

```
@@ -0,0 +1,119 @@
+        self.assertFalse(IProtectDossier(dossier).is_dossier_protected())
+
+    @browsing
+    def test_can_protect_dossier_on_post(self, browser):
+        self.login(self.dossier_manager, browser=browser)
+
+        data = {
+            '@type': 'opengever.dossier.businesscasedossier',
+            'title': 'Protected dossier',
+            'responsible': self.dossier_manager.getId(),
+            'reading': [self.regular_user.getId()],
+            'dossier_manager': self.dossier_manager.getId()
+        }
+
+        with self.observe_children(self.leaf_repofolder) as children:
+            browser.open(self.leaf_repofolder, data=json.dumps(data),
+                         method='POST', headers=self.api_headers)
+
+        self.assertEqual(201, browser.status_code)
+        self.assertEqual(1, len(children['added']))
+        dossier = children['added'].pop()
+
+        self.assertTrue(IProtectDossier(dossier).is_dossier_protected())
+
+    @browsing
+    def test_can_protect_dossier_on_patch(self, browser):
+        self.login(self.dossier_manager, browser=browser)
+
+        data = {
+            'reading': [self.regular_user.getId()],
+            'dossier_manager': self.dossier_manager.getId()
+        }
+
+        browser.open(self.dossier, data=json.dumps(data),
+                     method='PATCH', headers=self.api_headers)
+
+        self.assertTrue(IProtectDossier(self.dossier).is_dossier_protected())
+
+    @browsing
+    def test_only_dossier_manager_can_protect_dossier_on_post(self, browser):
+        self.login(self.regular_user, browser=browser)
+
+        data = {
+            '@type': 'opengever.dossier.businesscasedossier',
+            'title': 'Protected dossier',
+            'responsible': self.dossier_manager.getId(),
+            'reading': [self.regular_user.getId()],
+            'dossier_manager': self.dossier_manager.getId()
+        }
+
+        browser.open(self.leaf_repofolder, data=json.dumps(data),
```

Shouldn't that throw a validation error?

---

no, not through the plone restapi. Protected fields will just be ignored: https://github.com/plone/plone.restapi/blob/master/src/plone/restapi/deserializer/dxcontent.py#L98

---

I don't like all these things getting silently dropped, but we're not changing the RESTAPI... so ok